src/test/java/io/strimzi/kafka/bridge/http/HttpBridgeTest.java

```diff
@@ -1636,4 +1639,400 @@ public void doNotRespondTooLongMessage(TestContext context) {
+        Async consumeAsync = context.async();
+
+        client.get(BRIDGE_PORT, BRIDGE_HOST, baseUri + "/records" + "?timeout=" + String.valueOf(1000))
+                .as(BodyCodec.jsonArray())
+                .send(ar -> {
+                    context.assertTrue(ar.succeeded());
+
+                    JsonArray body = ar.result().body();
+                    context.assertEquals(10, body.size());
+                    consumeAsync.complete();
+                });
+
+        consumeAsync.await();
+
+        // seek
+        Async seekAsync = context.async();
+
+        JsonArray partitions = new JsonArray();
+        partitions.add(new JsonObject().put("topic", topic).put("partition", 0));
+
+        JsonObject root = new JsonObject();
+        root.put("partitions", partitions);
+
+        client.post(BRIDGE_PORT, BRIDGE_HOST, baseUri + "/positions/beginning")
+                .putHeader("Content-length", String.valueOf(root.toBuffer().length()))
+                .as(BodyCodec.jsonObject())
+                .sendJsonObject(root, ar -> {
+                    context.assertTrue(ar.succeeded());
+                    context.assertEquals(204, ar.result().statusCode());
+                    seekAsync.complete();
+                });
+
+        seekAsync.await();
+
+        // consume records
+        Async consumeSeekAsync = context.async();
+
+        client.get(BRIDGE_PORT, BRIDGE_HOST, baseUri + "/records")
+                .as(BodyCodec.jsonArray())
+                .send(ar -> {
+                    context.assertTrue(ar.succeeded());
+
+                    JsonArray body = ar.result().body();
+                    context.assertEquals(10, body.size());
+                    consumeSeekAsync.complete();
+                });
+
+        consumeSeekAsync.await();
+
+        // consumer deletion
+        Async deleteAsync = context.async();
+
+        client.delete(BRIDGE_PORT, BRIDGE_HOST, baseUri)
+                .as(BodyCodec.jsonObject())
+                .send(ar -> {
+                    context.assertTrue(ar.succeeded());
+
+                    HttpResponse<JsonObject> response = ar.result();
+                    context.assertEquals(ErrorCodeEnum.NO_CONTENT.getValue(), response.statusCode());
+
+                    deleteAsync.complete();
+                });
+
+        deleteAsync.await();
+    }
+
+    @Test
+    public void seekToEndAndReceive(TestContext context) {
+        String topic = "seekToEndAndReceive";
+        kafkaCluster.createTopic(topic, 1, 1);
+
+        Async creationAsync = context.async();
+
+        WebClient client = WebClient.create(vertx);
+
+        String name = "my-kafka-consumer";
+        String groupId = "my-group";
+
+        String baseUri = "http://" + BRIDGE_HOST + ":" + BRIDGE_PORT + "/consumers/" + groupId + "/instances/" + name;
+
+        JsonObject json = new JsonObject();
+        json.put("name", name);
+
+        client.post(BRIDGE_PORT, BRIDGE_HOST, "/consumers/" + groupId)
```

Comment:
You have all these repeated blocks of code for setting up the bridge so you can test it, _surely_ you can factor out a method for each HTTP request you're making?

Reply:
We already have #169 for this :-)